### PR TITLE
remove Ord UTxO constraint and make standalone derivations

### PR DESCRIPTION
**Pull request by polinavino**
Polymorphic TxOut will require standalone derivations as some cases
This is in preparation for making it polymorphic

### CODE REVIEW

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs]: Why `UseIsNormalFormNamed`? The downside of this compared to a generic instance is that when we find a thunk, its location will be `ShelleyGenesis`, instead of the precise field/type containing the thunk, e.g., `GenDelegPair`.

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxBody.hs
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxBody.hs]: Why not the generic instance?

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxBody.hs
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxBody.hs]: Why this boilerplate instead of the generic instance?

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxBody.hs
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxBody.hs]: Why this change?

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxBody.hs
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxBody.hs]: Why not derive it?

**Review by mrBliss — CHANGES_REQUESTED** [review]
Maybe I'm missing something, but many of these manual instance (and deriving via `UseIsNormalFormNamed`) can just be derived using standalone deriving + newtype/anyclass with the right context.

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs
Comment (polinavino) [reply shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs]: I am going to put back all the contentious derivations you commented on to the way they were in master for now, and then will change them back if/when needed for TxOut parametrization

Comment (mrBliss) [reply shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs]: Great! Thanks (There might be some more that I didn't comment on, but my opinion on them is the same :slightly_smiling_face:).

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs]: Not needed

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Chain.hs
Comment (mrBliss) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Chain.hs]: Not needed

**Review by mrBliss — APPROVED** [review]
LGTM, but maybe a member from the ledger team should approve too.

**Comment by mrBliss** [comment]
(Please squash before merging)

**Comment by mrBliss** [comment]
> (Please squash before merging)

Can we try not to fill the history with "ormolu" and "ormolise" commits? They don't add any value to the history and make `git blame` harder to use. Just squash them in the previous message (just like you would squash a typo, there's no value in keeping such a commit in your git history except to remind everyone that you made that typo) or better yet, let your editor run ormolu before saving :slightly_smiling_face: 

```
$ git log --oneline | grep ormo
275aa50a ormolise
3f34f6e3 ormolise
c3141472 ormolise
28fd134e ormolu
17e1fc32 ormolu
3ff5e9d8 ormolu
ff0ce463 ormolu
915bb386 ormolu
c7e36b6e ormolu
bef4ac9c ormolu
3ae45c6a ormolu
eeca9709 Running ormolu again
431031f6 Run ormolu
9dbe100e Run ormolu
5b6837af ormolu
e667c9b8 ormolu
...
```

**Pull request merged**
